### PR TITLE
add annotation for multiprocessing.{Value,Array} with special c-types

### PR DESCRIPTION
**Pull request by prauscher**
This PR adds support for `multiprocessing.Value` with standard c-types like c_float. Before this PR, Code like
```python
from multiprocessing import Value
from ctypes import c_float
a = Value(c_float, 0.0)
a.value = 1.2 
```
lead to `incompatible types in assignment`.

Also this PR resolves a Problem addressed in #4266 where
```python
from multiprocessing import Array
from ctypes import c_char
a = Array(c_char, 40)
print(a.value)
```
lead to `SynchronizedArray[c_char] has no attribute value`, altrough documentation states that such arrays have a `raw` and a `value` field.

Happy for your comments, lets see what I missed :)

### CODE REVIEW

File: stdlib/multiprocessing/context.pyi [hunk 1]
```diff
@@ -79,14 +80,22 @@ class BaseContext:
...
     @overload
-    def Value(self, typecode_or_type: type[_CT], *args: Any, lock: Literal[False]) -> Synchronized[_CT]: ...
+    def Value(
+        self, typecode_or_type: _SimpleCData[_T], *args: Any, lock: Literal[True] | _LockLike = True
+    ) -> Synchronized[_T]: ...
     @overload
-    def Value(self, typecode_or_type: type[_CT], *args: Any, lock: Literal[True] | _LockLike = True) -> Synchronized[_CT]: ...
+    def Value(self, typecode_or_type: _CT, *args: Any, lock: Literal[False]) -> Synchronized[_CT]: ...
+    @overload
+    def Value(self, typecode_or_type: _CT, *args: Any, lock: Literal[True] | _LockLike = True) -> Synchronized[_CT]: ...
```
Comment (srittau) [inline stdlib/multiprocessing/context.pyi:89]: You removed the `type` "wrapper" here, while `Array` still has it, was this intentional?

Also, could you add some test cases to `test_cases/stdlib`? I think the multiprocessing stubs are sufficiently complex to warrant some.

Comment (prauscher) [reply stdlib/multiprocessing/context.pyi:89]: this was intentional and a misstake, which is why i re-added them now. I agree with test-cases, let's see what I can come up with.

**Comment by prauscher** [comment]
What I could not get working:
```python
from multiprocessing import Array
from ctypes import c_float
a = Array(c_float, 2)
a[:] = [1.2, 4.1]  # Incompatible types: expect c_float, found float
```

Sadly, adding a new overload for this like it is done with `Value` breaks, so this requires a bit more thinking

File: test_cases/stdlib/check_multiprocessing.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+from ctypes import c_char, c_float
+from multiprocessing import Array, Value
+from multiprocessing.sharedctypes import Synchronized, SynchronizedString
+from typing_extensions import assert_type
+
+string = Array(c_char, 12)
+assert_type(string, SynchronizedString)
+
+field = Value(c_float, 0.0)
+assert_type(field, Synchronized[float])
```
Comment (srittau) [inline test_cases/stdlib/check_multiprocessing.py:12]:
```suggestion
assert_type(field, Synchronized[float])
a.value = 1.2
```

File: test_cases/stdlib/check_multiprocessing.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
+
+from ctypes import c_char, c_float
+from multiprocessing import Array, Value
+from multiprocessing.sharedctypes import Synchronized, SynchronizedString
+from typing_extensions import assert_type
+
+string = Array(c_char, 12)
+assert_type(string, SynchronizedString)
```
Comment (srittau) [inline test_cases/stdlib/check_multiprocessing.py:9]:
```suggestion
assert_type(string, SynchronizedString)
assert_type(a.value, bytes)
```

**Review by srittau — COMMENTED** [review]
I suggest to add the examples from your initial comment to the tests, see below.

File: test_cases/stdlib/check_multiprocessing.py (hunk 3 above)
Comment (prauscher) [reply test_cases/stdlib/check_multiprocessing.py:9]: Thanks, check is extended :)

File: test_cases/stdlib/check_multiprocessing.py (hunk 2 above)
Comment (prauscher) [reply test_cases/stdlib/check_multiprocessing.py:13]: Thanks, check is extended :)

**Review by srittau — APPROVED** [review]

**Pull request merged**